py/dml/c_backend.py

```diff
@@ -2266,6 +2271,16 @@ def init_trait_vtable(node, trait, param_overrides):
                     session_node,
                     (mkIntegerConstant(node.site, 0, False),)
                     * node.dimensions)))
+        else:
+            assert member_kind == 'memoized_outs'
+            typ = trait.vtable_trait(name).vtable_memoized_outs[name]
+            if node.dimensions:
+                for d in reversed(node.dimsizes):
+                    typ = TArray(typ, mkIntegerLiteral(node.site, d))
+                arg = f'calloc(1, sizeof({typ.declaration("")}))'
+            else:
+                arg = f'({{static {typ.declaration("_tmp")}; &_tmp; }})'
```

Comment:
why not unconditionally a `{static {TArray(typ, reduce(operator.mul, dimsizes, 1)).declaration("_tmp")}; _tmp}`?